src/Features/Core/Portable/ExternalAccess/UnitTesting/UnitTestingSolutionCrawlerServiceAccessor.cs

```diff
@@ -12,13 +12,47 @@ namespace Microsoft.CodeAnalysis.ExternalAccess.UnitTesting
 {
     internal sealed class UnitTestingSolutionCrawlerServiceAccessor : IUnitTestingSolutionCrawlerServiceAccessor
     {
-        private readonly ISolutionCrawlerService _implementation;
+        private readonly ISolutionCrawlerRegistrationService _registrationService;
+        private readonly ISolutionCrawlerService _solutionCrawlerService;
+
+        private UnitTestingIncrementalAnalyzerProvider _analyzerProvider;
 
         [Obsolete(MefConstruction.FactoryMethodMessage, error: true)]
-        public UnitTestingSolutionCrawlerServiceAccessor(ISolutionCrawlerService implementation)
-            => _implementation = implementation;
+        public UnitTestingSolutionCrawlerServiceAccessor(
+            ISolutionCrawlerRegistrationService registrationService,
+            ISolutionCrawlerService solutionCrawlerService)
+        {
+            _registrationService = registrationService;
+            _solutionCrawlerService = solutionCrawlerService;
+        }
+
+        public void AddAnalyzerProvider(IUnitTestingIncrementalAnalyzerProviderImplementation provider, UnitTestingIncrementalAnalyzerProviderMetadataWrapper metadata)
+        {
+            if (_analyzerProvider != null)
+            {
+                // NOTE: We expect the analyzer to be a singleton, therefore this method should be called just once.
+                throw new InvalidOperationException();
+            }
+
+            _analyzerProvider = new UnitTestingIncrementalAnalyzerProvider(provider);
+            _registrationService.AddAnalyzerProvider(_analyzerProvider, metadata.UnderlyingObject);
+        }
+
+        // NOTE: For the Reanalyze method to work correctly, the analyzer passed into the Reanalyze method,
+        //       must be the same as created when we call the AddAnalyzerProvider method.
+        //       As such the analyzer provider instance caches a single instance of the analyzer.
+        public void Reanalyze(Workspace workspace, IEnumerable<ProjectId> projectIds = null, IEnumerable<DocumentId> documentIds = null, bool highPriority = false)
```

Comment:
💡 It should not be necessary to duplicate the default arguments here.

```suggestion
        public void Reanalyze(Workspace workspace, IEnumerable<ProjectId> projectIds, IEnumerable<DocumentId> documentIds, bool highPriority)
```

Reply:
+1 - this should go away if we only keep the required arguments though...